src/app/webui/components/hooks/useChat.ts

```diff
@@ -0,0 +1,249 @@
+import { useState, useRef, useEffect, useCallback } from 'react';
+
+// Define the structure for message content parts
+export interface TextPart {
+    type: 'text';
+    text: string;
+}
+
+export interface ImagePart {
+    type: 'image';
+    // Include relevant properties from backend ImageData if needed for display
+    base64: string;
+    mimeType: string;
+}
+
+export interface Message {
+    id: string;
+    role: 'user' | 'system' | 'assistant' | 'tool';
+    createdAt: number;
+    // Content for text-based messages
+    content: string | null | Array<TextPart | ImagePart>;
+    // Optional image data sent directly in user messages
+    imageData?: { base64: string; mimeType: string };
+    // Tool call and result metadata
+    toolName?: string;
+    toolArgs?: any;
+    toolResult?: any;
+}
+
```

Comment:
can we re-use these types from core

@coderabbitai check if we have types like this anywhere in core and if we can/should use those

Reply:
yes let's do this so that the types stay in sync

also update the import to use @core instead of ../../.. relative import for modularity